server.js

```diff
@@ -1,38 +1,159 @@
+import netflixData from './data/netflix-titles.json';
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
 // import goldenGlobesData from './data/golden-globes.json'
 // import avocadoSalesData from './data/avocado-sales.json'
 // import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
+
 // import topMusicData from './data/top-music.json'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
+const mongoUrl =
+	process.env.MONGO_URL || 'mongodb://localhost/project-mongo-netflix';
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// Defines the port the app will run on
+const port = process.env.PORT || 8080;
+const app = express();
+
+//A show database model to start populating the database
+//shows all the existing data
+const Show = mongoose.model('Show', {
+	show_id: Number,
+	title: String,
+	director: String,
+	cast: String,
+	country: String,
+	date_added: String,
+	release_year: Number,
+	rating: String,
+	duration: String,
+	listed_in: String,
+	description: String,
+	type: String,
+});
+
+//Function to start seeding the database,
+//and it will only run when variable is present and true
+if (process.env.RESET_DB) {
+	console.log('Resetting database!');
+	const seedDataBase = async () => {
+		//Deletes pre-existing shows to prevent duplicates
+		await Show.deleteMany({});
+
+		//Creates a new show
+		netflixData.forEach((item) => {
+			const newShow = new Show(item);
+			newShow.save();
+		});
+	};
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+	seedDataBase();
+}
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(express.json())
+app.use(cors());
+app.use(express.json());
+
+// Our own middlewarethat checks if the database is connected before going to our endpoints
+app.use((req, res, next) => {
+	if (mongoose.connection.readyState === 1) {
+		next();
+	} else {
+		res.status(503).json({ error: 'Service unavailable' });
+	}
+});
+
+//RESTful route
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+	res.send('Hello world, welcome to netflix titles API by Madelene Trang');
+});
+
+//showing all the possible enpoints in the app
+app.get('/endpoints', (req, res) => {
+	res.send(listEndpoints(app));
+});
+
+//----COLLECTION 0F RESULT------
+//endpoint to get all shows
+//Query params
+app.get('/shows', async (req, res) => {
+	console.log(req.query);
+
+	const { type, country, release_year, director } = req.query;
+	let showsToSend = await Show.find(req.query);
+
```

Comment:
I think we discussed this in the group meeting (and I know you have had HUGE problems with Heroku), but as far as I understand, if you have (req.query) in line 89, then lines 91-110 are superfluous. This, in my opinion, is a nice feature of the Mongoose framework. BUT I understand that you don't want to change the code too much due to your difficulties